worlds/zork_grand_inquisitor/data_funcs.py

```diff
@@ -0,0 +1,256 @@
+from typing import Dict, Set, Tuple, Union
+
+from .data.entrance_rule_data import entrance_rule_data
+from .data.item_data import item_data, ZorkGrandInquisitorItemData
+from .data.location_data import location_data, ZorkGrandInquisitorLocationData
+
+from .enums import (
+    ZorkGrandInquisitorEvents,
+    ZorkGrandInquisitorGoals,
+    ZorkGrandInquisitorItems,
+    ZorkGrandInquisitorLocations,
+    ZorkGrandInquisitorRegions,
+    ZorkGrandInquisitorTags,
+)
+
+
+def item_names_to_id() -> Dict[str, int]:
+    return {item.value: data.archipelago_id for item, data in item_data.items()}
+
+
+def item_names_to_item() -> Dict[str, ZorkGrandInquisitorItems]:
+    return {item.value: item for item, _ in item_data.items()}
+
+
+def location_names_to_id() -> Dict[str, int]:
+    return {
+        location.value: data.archipelago_id
+        for location, data in location_data.items()
+        if data.archipelago_id is not None
+    }
+
+
+def location_names_to_location() -> Dict[str, ZorkGrandInquisitorLocations]:
+    return {
+        location.value: location
+        for location, data in location_data.items()
+        if data.archipelago_id is not None
+    }
+
+
+def id_to_goals() -> Dict[int, ZorkGrandInquisitorGoals]:
+    return {goal.value: goal for goal in ZorkGrandInquisitorGoals}
+
+
+def id_to_items() -> Dict[int, ZorkGrandInquisitorItems]:
+    return {data.archipelago_id: item for item, data in item_data.items()}
+
+
+def id_to_locations() -> Dict[int, ZorkGrandInquisitorLocations]:
+    return {
+        data.archipelago_id: location
+        for location, data in location_data.items()
+        if data.archipelago_id is not None
+    }
+
+
+def item_groups() -> Dict[str, Set[str]]:
+    groups: Dict[str, Set[str]] = dict()
+
+    tag: ZorkGrandInquisitorTags
+    for tag in ZorkGrandInquisitorTags:
+        groups[tag.value] = set()
+
+    item: ZorkGrandInquisitorItems
+    data: ZorkGrandInquisitorItemData
+    for item, data in item_data.items():
+        if data.tags is not None:
+            for tag in data.tags:
+                groups[tag.value].add(item.value)
```

Comment:
this allows you to remove the tag loop above
```suggestion
                groups.setdefault(tag.value, set()).add(item.value)
```